aws_networkfirewall_firewall_policy data lookup fails with `Error: setting firewall_policy: Invalid address to set: []string{"firewall_policy", "0", "policy_variables"}`

### Terraform and AWS Provider Version

```console
Terraform v1.11.4
on darwin_arm64
+ provider registry.terraform.io/hashicorp/aws v5.96.0
```

### Affected Resource(s) or Data Source(s)

- `data-sources/networkfirewall_firewall_policy`

### Expected Behavior

I expect that performing a data lookup by name or ARN on an existing `networkfirewall_firewall_policy` resource should succeed.

### Actual Behavior

Attempting to do a data lookup on any existing `networkfirewall_firewall_policy` is returning an error and failing to return any information about the resource.

### Relevant Error/Panic Output

```console
Error: setting firewall_policy: Invalid address to set: []string{"firewall_policy", "0", "policy_variables"}
```


### Sample Terraform Configuration

<details>
<summary>Click to expand configuration</summary>

```hcl
data "aws_networkfirewall_firewall_policy" "fms_managed" {
  name = "test"
}
```
</details>


### Steps to Reproduce

1. Create a Network Firewall with any configuration.
2. Update the data lookup to use the name of the firewall you created.
3. Run a terraform plan.
4. Receive an error.

### Debug Logging

<details>
<summary>Click to expand log output</summary>

```console
aws-vault exec entnet -- terraform plan
[INFO]  Terraform version: 1.11.4
[DEBUG] using github.com/hashicorp/go-tfe v1.70.0
[DEBUG] using github.com/hashicorp/hcl/v2 v2.23.0
[DEBUG] using github.com/hashicorp/terraform-svchost v0.1.1
[DEBUG] using github.com/zclconf/go-cty v1.16.0
[INFO]  Go runtime version: go1.23.3
[INFO]  CLI args: []string{"terraform", "plan"}
[DEBUG] Attempting to open CLI config file: /Users/user1/.terraformrc
[DEBUG] File doesn't exist, but doesn't need to. Ignoring.
[DEBUG] ignoring non-existing provider search directory terraform.d/plugins
[DEBUG] ignoring non-existing provider search directory /Users/user1/.terraform.d/plugins
[DEBUG] ignoring non-existing provider search directory /Users/user1/Library/Application Support/io.terraform/plugins
[DEBUG] ignoring non-existing provider search directory /Library/Application Support/io.terraform/plugins
[INFO]  CLI command args: []string{"plan"}
[DEBUG] checking for provisioner in "."
[DEBUG] checking for provisioner in "/users/user1/bin"
[INFO]  backend/local: starting Plan operation
[DEBUG] created provider logger: level=debug
[INFO]  provider: configuring client automatic mTLS
[DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.96.0/darwin_arm64/terraform-provider-aws_v5.96.0_x5 args=[".terraform/providers/registry.terraform.io/hashicorp/aws/5.96.0/darwin_arm64/terraform-provider-aws_v5.96.0_x5"]
[DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.96.0/darwin_arm64/terraform-provider-aws_v5.96.0_x5 pid=34552
[DEBUG] provider: waiting for RPC address: plugin=.terraform/providers/registry.terraform.io/hashicorp/aws/5.96.0/darwin_arm64/terraform-provider-aws_v5.96.0_x5
[DEBUG] provider.terraform-provider-aws_v5.96.0_x5: Starting github.com/hashicorp/terraform-provider-aws@5.96.0 (go1.23.8)...
[DEBUG] provider.terraform-provider-aws_v5.96.0_x5: Initializing Terraform AWS Provider...
[INFO]  provider.terraform-provider-aws_v5.96.0_x5: configuring server automatic mTLS: timestamp=2025-05-01T11:08:35.393-0700
[DEBUG] provider.terraform-provider-aws_v5.96.0_x5: plugin address: address=/var/folders/m0/vj5jknf1053fy6f01xshcx440000gn/T/plugin1702241856 network=unix timestamp=2025-05-01T11:08:35.398-0700
[DEBUG] provider: using plugin: version=5
[DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
[INFO]  provider: plugin process exited: plugin=.terraform/providers/registry.terraform.io/hashicorp/aws/5.96.0/darwin_arm64/terraform-provider-aws_v5.96.0_x5 id=34552
[DEBUG] provider: plugin exited
[DEBUG] Building and walking validate graph
[DEBUG] adding implicit provider configuration provider["registry.terraform.io/hashicorp/aws"], implied first by data.aws_networkfirewall_firewall_policy.fms_managed
[DEBUG] ProviderTransformer: "data.aws_networkfirewall_firewall_policy.fms_managed" (*terraform.NodeValidatableResource) needs provider["registry.terraform.io/hashicorp/aws"]
DEBUG] ReferenceTransformer: "data.aws_networkfirewall_firewall_policy.fms_managed" references: []
[DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/hashicorp/aws\"]" references: []
[DEBUG] Starting graph walk: walkValidate
[DEBUG] created provider logger: level=debug
[INFO]  provider: configuring client automatic mTLS
[DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.96.0/darwin_arm64/terraform-provider-aws_v5.96.0_x5 args=[".terraform/providers/registry.terraform.io/hashicorp/aws/5.96.0/darwin_arm64/terraform-provider-aws_v5.96.0_x5"]
[DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.96.0/darwin_arm64/terraform-provider-aws_v5.96.0_x5 pid=34554
[DEBUG] provider: waiting for RPC address: plugin=.terraform/providers/registry.terraform.io/hashicorp/aws/5.96.0/darwin_arm64/terraform-provider-aws_v5.96.0_x5
[DEBUG] provider.terraform-provider-aws_v5.96.0_x5: Starting github.com/hashicorp/terraform-provider-aws@5.96.0 (go1.23.8)...
[DEBUG] provider.terraform-provider-aws_v5.96.0_x5: Initializing Terraform AWS Provider...
[INFO]  provider.terraform-provider-aws_v5.96.0_x5: configuring server automatic mTLS: timestamp=2025-05-01T11:08:35.963-0700
[DEBUG] provider.terraform-provider-aws_v5.96.0_x5: plugin address: network=unix address=/var/folders/m0/vj5jknf1053fy6f01xshcx440000gn/T/plugin935006329 timestamp=2025-05-01T11:08:35.968-0700
[DEBUG] provider: using plugin: version=5
[DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
[INFO]  provider: plugin process exited: plugin=.terraform/providers/registry.terraform.io/hashicorp/aws/5.96.0/darwin_arm64/terraform-provider-aws_v5.96.0_x5 id=34554
[DEBUG] provider: plugin exited
[INFO]  backend/local: plan calling Plan
[DEBUG] Building and walking plan graph for NormalMode
[DEBUG] adding implicit provider configuration provider["registry.terraform.io/hashicorp/aws"], implied first by data.aws_networkfirewall_firewall_policy.fms_managed (expand)
DEBUG] ProviderTransformer: "data.aws_networkfirewall_firewall_policy.fms_managed (expand)" (*terraform.nodeExpandPlannableResource) needs provider["registry.terraform.io/hashicorp/aws"]
[DEBUG] ReferenceTransformer: "data.aws_networkfirewall_firewall_policy.fms_managed (expand)" references: []
[DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/hashicorp/aws\"]" references: []
[DEBUG] Starting graph walk: walkPlan
[DEBUG] created provider logger: level=debug
[INFO]  provider: configuring client automatic mTLS
[DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.96.0/darwin_arm64/terraform-provider-aws_v5.96.0_x5 args=[".terraform/providers/registry.terraform.io/hashicorp/aws/5.96.0/darwin_arm64/terraform-provider-aws_v5.96.0_x5"]
[DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.96.0/darwin_arm64/terraform-provider-aws_v5.96.0_x5 pid=34555
[DEBUG] provider: waiting for RPC address: plugin=.terraform/providers/registry.terraform.io/hashicorp/aws/5.96.0/darwin_arm64/terraform-provider-aws_v5.96.0_x5
[DEBUG] provider.terraform-provider-aws_v5.96.0_x5: Starting github.com/hashicorp/terraform-provider-aws@5.96.0 (go1.23.8)...
[DEBUG] provider.terraform-provider-aws_v5.96.0_x5: Initializing Terraform AWS Provider...
[INFO]  provider.terraform-provider-aws_v5.96.0_x5: configuring server automatic mTLS: timestamp=2025-05-01T11:08:36.093-0700
[DEBUG] provider.terraform-provider-aws_v5.96.0_x5: plugin address: address=/var/folders/m0/vj5jknf1053fy6f01xshcx440000gn/T/plugin3128765422 network=unix timestamp=2025-05-01T11:08:36.097-0700
[DEBUG] provider: using plugin: version=5
[DEBUG] provider.terraform-provider-aws_v5.96.0_x5: Configuring Terraform AWS Provider: @caller=github.com/hashicorp/terraform-provider-aws/internal/conns/config.go:144 tf_rpc=ConfigureProvider tf_req_id=ed753eb6-05cd-1b8c-3006-7a0da1355bdd @module=aws tf_mux_provider="*schema.GRPCProviderServer" tf_provider_addr=registry.terraform.io/hashicorp/aws timestamp=2025-05-01T11:08:36.145-0700
[DEBUG] provider.terraform-provider-aws_v5.96.0_x5: Resolving credentials provider: tf_req_id=ed753eb6-05cd-1b8c-3006-7a0da1355bdd tf_rpc=ConfigureProvider @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.64/logging/tf_logger.go:47 tf_provider_addr=registry.terraform.io/hashicorp/aws @module=aws.aws-base tf_mux_provider="*schema.GRPCProviderServer" timestamp=2025-05-01T11:08:36.145-0700
[DEBUG] provider.terraform-provider-aws_v5.96.0_x5: Loading configuration: @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.64/logging/tf_logger.go:47 tf_provider_addr=registry.terraform.io/hashicorp/aws tf_req_id=ed753eb6-05cd-1b8c-3006-7a0da1355bdd @module=aws.aws-base tf_mux_provider="*schema.GRPCProviderServer" tf_rpc=ConfigureProvider timestamp=2025-05-01T11:08:36.145-0700
[DEBUG] provider.terraform-provider-aws_v5.96.0_x5: Retrieving credentials: tf_req_id=ed753eb6-05cd-1b8c-3006-7a0da1355bdd tf_rpc=ConfigureProvider tf_mux_provider="*schema.GRPCProviderServer" tf_provider_addr=registry.terraform.io/hashicorp/aws @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.64/logging/tf_logger.go:47 @module=aws.aws-base timestamp=2025-05-01T11:08:36.146-0700
[INFO]  provider.terraform-provider-aws_v5.96.0_x5: Retrieved credentials: @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.64/logging/tf_logger.go:39 tf_req_id=ed753eb6-05cd-1b8c-3006-7a0da1355bdd tf_rpc=ConfigureProvider @module=aws.aws-base tf_aws.credentials_source=EnvConfigCredentials tf_mux_provider="*schema.GRPCProviderServer" tf_provider_addr=registry.terraform.io/hashicorp/aws timestamp=2025-05-01T11:08:36.146-0700
[DEBUG] provider.terraform-provider-aws_v5.96.0_x5: Loading configuration: @module=aws.aws-base tf_provider_addr=registry.terraform.io/hashicorp/aws @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.64/logging/tf_logger.go:47 tf_mux_provider="*schema.GRPCProviderServer" tf_req_id=ed753eb6-05cd-1b8c-3006-7a0da1355bdd tf_rpc=ConfigureProvider timestamp=2025-05-01T11:08:36.146-0700
[DEBUG] provider.terraform-provider-aws_v5.96.0_x5: Creating AWS SDK v1 session: @caller=github.com/hashicorp/terraform-provider-aws/internal/conns/config.go:168 tf_rpc=ConfigureProvider @module=aws tf_mux_provider="*schema.GRPCProviderServer" tf_provider_addr=registry.terraform.io/hashicorp/aws tf_req_id=ed753eb6-05cd-1b8c-3006-7a0da1355bdd timestamp=2025-05-01T11:08:36.146-0700
[DEBUG] provider.terraform-provider-aws_v5.96.0_x5: Retrieving AWS account details: tf_mux_provider="*schema.GRPCProviderServer" tf_rpc=ConfigureProvider @caller=github.com/hashicorp/terraform-provider-aws/internal/conns/config.go:183 @module=aws tf_provider_addr=registry.terraform.io/hashicorp/aws tf_req_id=ed753eb6-05cd-1b8c-3006-7a0da1355bdd timestamp=2025-05-01T11:08:36.146-0700
[DEBUG] provider.terraform-provider-aws_v5.96.0_x5: Retrieving caller identity from STS: tf_req_id=ed753eb6-05cd-1b8c-3006-7a0da1355bdd tf_rpc=ConfigureProvider @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.64/logging/tf_logger.go:47 @module=aws.aws-base tf_mux_provider="*schema.GRPCProviderServer" tf_provider_addr=registry.terraform.io/hashicorp/aws timestamp=2025-05-01T11:08:36.146-0700
[DEBUG] provider.terraform-provider-aws_v5.96.0_x5: HTTP Request Sent: http.method=POST http.request.header.amz_sdk_invocation_id=71690718-efb8-482a-9d69-ffe83309f1f5 tf_aws.sdk=aws-sdk-go-v2 tf_aws.signing_region="" tf_mux_provider="*schema.GRPCProviderServer" tf_rpc=ConfigureProvider aws.region=us-east-1 http.request.header.content_type=application/x-www-form-urlencoded http.request.header.authorization="AWS4-HMAC-SHA256 Credential=ASIA************ENCZ/20250501/us-east-1/sts/aws4_request, SignedHeaders=amz-sdk-invocation-id;amz-sdk-request;content-length;content-type;host;x-amz-date;x-amz-security-token, Signature=*****" tf_provider_addr=registry.terraform.io/hashicorp/aws net.peer.name=sts.us-east-1.amazonaws.com rpc.method=GetCallerIdentity
  http.request.body=
  | Action=GetCallerIdentity&Version=2011-06-15
   http.request.header.amz_sdk_request="attempt=1; max=25" http.request.header.x_amz_security_token="*****" http.user_agent="APN/1.0 HashiCorp/1.0 Terraform/1.11.4 (+https://www.terraform.io) terraform-provider-aws/5.96.0 (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go-v2/1.36.3 ua/2.1 os/macos lang/go#1.23.8 md/GOOS#darwin md/GOARCH#arm64 api/sts#1.33.19 m/g" rpc.system=aws-api http.request_content_length=43 tf_req_id=ed753eb6-05cd-1b8c-3006-7a0da1355bdd @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.64/logging/tf_logger.go:47 @module=aws.aws-base http.request.header.x_amz_date=20250501T180836Z http.url=https://sts.us-east-1.amazonaws.com/ rpc.service=STS timestamp=2025-05-01T11:08:36.147-0700
[DEBUG] provider.terraform-provider-aws_v5.96.0_x5: HTTP Response Received:
  http.response.body=
  | <GetCallerIdentityResponse xmlns="https://sts.amazonaws.com/doc/2011-06-15/">
  |   <GetCallerIdentityResult>
  |     <Arn>arn:aws:sts::AWSACCTID:assumed-role/AWSReservedSSO_admin_000000000000/user1@domain.xyz</Arn>
  |     <UserId>AROA*************DZVL:user1@domain.xyz</UserId>
  |     <Account>AWSACCTID</Account>
  |   </GetCallerIdentityResult>
  |   <ResponseMetadata>
  |     <RequestId>3b8c9297-ff1e-4a45-9a39-68c2dafa86c6</RequestId>
  |   </ResponseMetadata>
  | </GetCallerIdentityResponse>
   http.response.header.x_amzn_requestid=3b8c9297-ff1e-4a45-9a39-68c2dafa86c6 http.response_content_length=496 tf_rpc=ConfigureProvider @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.64/logging/tf_logger.go:47 http.status_code=200 rpc.method=GetCallerIdentity rpc.service=STS tf_provider_addr=registry.terraform.io/hashicorp/aws tf_mux_provider="*schema.GRPCProviderServer" tf_req_id=ed753eb6-05cd-1b8c-3006-7a0da1355bdd @module=aws.aws-base aws.region=us-east-1 tf_aws.sdk=aws-sdk-go-v2 tf_aws.signing_region="" rpc.system=aws-api http.duration=366 http.response.header.content_type=text/xml http.response.header.date="Thu, 01 May 2025 18:08:36 GMT" timestamp=2025-05-01T11:08:36.513-0700
[INFO]  provider.terraform-provider-aws_v5.96.0_x5: Retrieved caller identity from STS: @module=aws.aws-base tf_provider_addr=registry.terraform.io/hashicorp/aws tf_req_id=ed753eb6-05cd-1b8c-3006-7a0da1355bdd tf_rpc=ConfigureProvider @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.64/logging/tf_logger.go:39 tf_mux_provider="*schema.GRPCProviderServer" timestamp=2025-05-01T11:08:36.514-0700
[DEBUG] Resource instance state not found for node "data.aws_networkfirewall_firewall_policy.fms_managed", instance data.aws_networkfirewall_firewall_policy.fms_managed
[DEBUG] ReferenceTransformer: "data.aws_networkfirewall_firewall_policy.fms_managed" references: []
data.aws_networkfirewall_firewall_policy.fms_managed: Reading...
[DEBUG] provider.terraform-provider-aws_v5.96.0_x5: HTTP Request Sent: rpc.service="Network Firewall" http.request.header.authorization="AWS4-HMAC-SHA256 Credential=ASIA************ENCZ/20250501/us-east-1/network-firewall/aws4_request, SignedHeaders=amz-sdk-invocation-id;amz-sdk-request;content-length;content-type;host;x-amz-date;x-amz-security-token;x-amz-target, Signature=*****" http.request.header.amz_sdk_invocation_id=2fcf6456-6d95-40f1-bda9-31b35fcbaa11 rpc.method=DescribeFirewallPolicy aws.region=us-east-1 http.method=POST http.request_content_length=29 @module=aws http.request.header.x_amz_date=20250501T180836Z http.request.header.x_amz_security_token="*****" http.user_agent="APN/1.0 HashiCorp/1.0 Terraform/1.11.4 (+https://www.terraform.io) terraform-provider-aws/5.96.0 (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go-v2/1.36.3 ua/2.1 os/macos lang/go#1.23.8 md/GOOS#darwin md/GOARCH#arm64 api/networkfirewall#1.47.2 m/g" net.peer.name=network-firewall.us-east-1.amazonaws.com tf_data_source_type=aws_networkfirewall_firewall_policy tf_mux_provider="*schema.GRPCProviderServer" tf_req_id=bcdba4e2-54b1-2a96-b5a3-66bb81f02316 http.request.header.amz_sdk_request="attempt=1; max=25" http.request.header.x_amz_target=NetworkFirewall_20201112.DescribeFirewallPolicy rpc.system=aws-api @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.64/logging/tf_logger.go:45 tf_aws.sdk=aws-sdk-go-v2 tf_provider_addr=registry.terraform.io/hashicorp/aws http.request.header.content_type=application/x-amz-json-1.0 tf_aws.signing_region="" tf_rpc=ReadDataSource
  http.request.body=
  | {"FirewallPolicyName":"test"}
   http.url=https://network-firewall.us-east-1.amazonaws.com/ timestamp=2025-05-01T11:08:36.531-0700
[DEBUG] provider.terraform-provider-aws_v5.96.0_x5: HTTP Response Received: @module=aws http.status_code=200 rpc.method=DescribeFirewallPolicy @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.64/logging/tf_logger.go:45 rpc.service="Network Firewall" rpc.system=aws-api tf_provider_addr=registry.terraform.io/hashicorp/aws tf_req_id=bcdba4e2-54b1-2a96-b5a3-66bb81f02316 aws.region=us-east-1 http.duration=446 http.response.header.content_type=application/x-amz-json-1.0 http.response.header.date="Thu, 01 May 2025 18:08:36 GMT" tf_mux_provider="*schema.GRPCProviderServer" tf_rpc=ReadDataSource
  http.response.body=
  | {"FirewallPolicy":{"PolicyVariables":{"RuleVariables":{}},"StatefulDefaultActions":["aws:alert_established","aws:drop_established"],"StatefulEngineOptions":{"RuleOrder":"STRICT_ORDER","StreamExceptionPolicy":"REJECT"},"StatefulRuleGroupReferences":[],"StatelessCustomActions":[],"StatelessDefaultActions":["aws:forward_to_sfe"],"StatelessFragmentDefaultActions":["aws:forward_to_sfe"],"StatelessRuleGroupReferences":[]},"FirewallPolicyResponse":{"ConsumedStatefulRuleCapacity":0,"ConsumedStatelessRuleCapacity":0,"Description":"test","EncryptionConfiguration":{"KeyId":"AWS_OWNED_KMS_KEY","Type":"AWS_OWNED_KMS_KEY"},"FirewallPolicyArn":"arn:aws:network-firewall:us-east-1:AWSACCTID:firewall-policy/test","FirewallPolicyId":"a8000006-7000-4004-b000-059f1ff5e5d8","FirewallPolicyName":"test","FirewallPolicyStatus":"ACTIVE","LastModifiedTime":1.746122332548E9,"NumberOfAssociations":0,"Tags":[]},"UpdateToken":"014fab88-7696-41bf-9e6b-42cffd9b12b2"}
   http.response.header.x_amzn_requestid=a4f7d8c1-099f-4045-b886-16ded1132436 http.response_content_length=951 tf_aws.sdk=aws-sdk-go-v2 tf_aws.signing_region="" tf_data_source_type=aws_networkfirewall_firewall_policy timestamp=2025-05-01T11:08:36.978-0700
[ERROR] provider.terraform-provider-aws_v5.96.0_x5: [ERROR] setting state: Invalid address to set: []string{"firewall_policy", "0", "policy_variables"}
[ERROR] provider.terraform-provider-aws_v5.96.0_x5: Response contains error diagnostic: tf_data_source_type=aws_networkfirewall_firewall_policy tf_proto_version=5.8 @module=sdk.proto diagnostic_severity=ERROR diagnostic_detail="" diagnostic_summary="setting firewall_policy: Invalid address to set: []string{\"firewall_policy\", \"0\", \"policy_variables\"}" tf_provider_addr=registry.terraform.io/hashicorp/aws tf_req_id=bcdba4e2-54b1-2a96-b5a3-66bb81f02316 tf_rpc=ReadDataSource @caller=github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/internal/diag/diagnostics.go:58 timestamp=2025-05-01T11:08:36.980-0700
[ERROR] vertex "data.aws_networkfirewall_firewall_policy.fms_managed" error: setting firewall_policy: Invalid address to set: []string{"firewall_policy", "0", "policy_variables"}
[ERROR] vertex "data.aws_networkfirewall_firewall_policy.fms_managed (expand)" error: setting firewall_policy: Invalid address to set: []string{"firewall_policy", "0", "policy_variables"}
[WARN]  Planning encountered errors, so plan is not applyable
[INFO]  backend/local: plan operation completed

Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: setting firewall_policy: Invalid address to set: []string{"firewall_policy", "0", "policy_variables"}
│
│   with data.aws_networkfirewall_firewall_policy.fms_managed,
│   on test.tf line 1, in data "aws_networkfirewall_firewall_policy" "fms_managed":
│    1: data "aws_networkfirewall_firewall_policy" "fms_managed" {
│
╵
[DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
[INFO]  provider: plugin process exited: plugin=.terraform/providers/registry.terraform.io/hashicorp/aws/5.96.0/darwin_arm64/terraform-provider-aws_v5.96.0_x5 id=34555
[DEBUG] provider: plugin exited

```
</details>


### GenAI / LLM Assisted Development

n/a

### Important Facts and References

_No response_

### Would you like to implement a fix?

No

--- a/.changelog/42473.txt
+++ b/.changelog/42473.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+data-source/aws_networkfirewall_firewall_policy: Add `firewall_policy.policy_variables` configuration block
+```

--- a/internal/service/networkfirewall/firewall_policy_data_source.go
+++ b/internal/service/networkfirewall/firewall_policy_data_source.go
@@ -43,6 +43,39 @@ func dataSourceFirewallPolicy() *schema.Resource {
 					Computed: true,
 					Elem: &schema.Resource{
 						Schema: map[string]*schema.Schema{
+							"policy_variables": {
+								Type:     schema.TypeList,
+								Computed: true,
+								Elem: &schema.Resource{
+									Schema: map[string]*schema.Schema{
+										"rule_variables": {
+											Type:     schema.TypeSet,
+											Computed: true,
+											Elem: &schema.Resource{
+												Schema: map[string]*schema.Schema{
+													"ip_set": {
+														Type:     schema.TypeList,
+														Computed: true,
+														Elem: &schema.Resource{
+															Schema: map[string]*schema.Schema{
+																"definition": {
+																	Type:     schema.TypeSet,
+																	Computed: true,
+																	Elem:     &schema.Schema{Type: schema.TypeString},
+																},
+															},
+														},
+													},
+													names.AttrKey: {
+														Type:     schema.TypeString,
+														Computed: true,
+													},
+												},
+											},
+										},
+									},
+								},
+							},
 							"stateful_default_actions": {
 								Type:     schema.TypeSet,
 								Computed: true,

--- a/internal/service/networkfirewall/firewall_policy_data_source_test.go
+++ b/internal/service/networkfirewall/firewall_policy_data_source_test.go
@@ -134,6 +134,34 @@ func TestAccNetworkFirewallFirewallPolicyDataSource_withOverriddenManagedRuleGro
 	})
 }
 
+func TestAccNetworkFirewallFirewallPolicyDataSource_withPolicyVariables(t *testing.T) {
+	ctx := acctest.Context(t)
+	rName := sdkacctest.RandomWithPrefix("resource-test-terraform")
+	resourceName := "aws_networkfirewall_firewall_policy.test"
+	datasourceName := "data.aws_networkfirewall_firewall_policy.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(ctx, t); testAccPreCheck(ctx, t) },
+		ErrorCheck:               acctest.ErrorCheck(t, names.NetworkFirewallServiceID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccFirewallPolicyDataSourceConfig_withPolicyVariables(rName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrPair(datasourceName, names.AttrARN, resourceName, names.AttrARN),
+					resource.TestCheckResourceAttrPair(datasourceName, names.AttrDescription, resourceName, names.AttrDescription),
+					resource.TestCheckResourceAttrPair(datasourceName, "firewall_policy.#", resourceName, "firewall_policy.#"),
+					resource.TestCheckResourceAttrPair(datasourceName, "firewall_policy.rule_variables.#", resourceName, "firewall_policy.rule_variables.#"),
+					resource.TestCheckResourceAttrPair(datasourceName, "firewall_policy.rule_variables.ip_set.#", resourceName, "firewall_policy.rule_variables.ip_set.#"),
+					resource.TestCheckResourceAttrPair(datasourceName, "firewall_policy.rule_variables.ip_set.0.definition.#", resourceName, "firewall_policy.rule_variables.ip_set.0.definition.#"),
+					resource.TestCheckResourceAttrPair(datasourceName, names.AttrName, resourceName, names.AttrName),
+					resource.TestCheckResourceAttrPair(datasourceName, acctest.CtTagsPercent, resourceName, acctest.CtTagsPercent),
+				),
+			},
+		},
+	})
+}
+
 func testAccFirewallPolicyDataSourceConfig_basic(rName string) string {
 	return fmt.Sprintf(`
 resource "aws_networkfirewall_firewall_policy" "test" {
@@ -192,6 +220,29 @@ resource "aws_networkfirewall_firewall_policy" "test" {
 
       override {
         action = "DROP_TO_ALERT"
+      }
+    }
+  }
+}
+
+data "aws_networkfirewall_firewall_policy" "test" {
+  arn = aws_networkfirewall_firewall_policy.test.arn
+}`, rName)
+}
+
+func testAccFirewallPolicyDataSourceConfig_withPolicyVariables(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_networkfirewall_firewall_policy" "test" {
+  name = %[1]q
+  firewall_policy {
+    stateless_fragment_default_actions = ["aws:drop"]
+    stateless_default_actions          = ["aws:pass"]
+    policy_variables {
+      rule_variables {
+        key = "HOME_NET"
+        ip_set {
+          definition = ["10.0.0.0/16", "10.1.0.0/24"]
+        }
       }
     }
   }

